code/modules/mob/living/simple_animal/bot/mulebot.dm

```diff
@@ -545,11 +541,19 @@
 
 /mob/living/simple_animal/bot/mulebot/Moved() //make sure we always use power after moving.
 	. = ..()
-	if(!cell)
-		return
-	cell.use(cell_move_power_usage)
+
+	cell?.use(cell_move_power_usage)
+
+	for(var/potential_target in loc.contents)
```

Comment:
This seems costly to run every single time they move. Why not have human/crossed() send a signal (COMSIG_CROSSED_HUMAN) and have mulebots listen for that instead?

Reply:
How would the Mulebot register that signal? It doesn't know about any potential targets to crush until it has actually entered the turf containing them.

Reply:
Really we need a signal or proc for when an object moves over another object.

Reply:
Though that might be significantly more costly.

Reply:
The problem with signals is you need to RegisterSignal on something - They're only useful if you know in advance the something that you want to listen to signals for. You could possibly RegisterSignal on every single human in GLOB.mob_living_list for every Mulebot, but that seems like a nightmare and you'd have to keep track of every new human created.

Reply:
Moved to signals! The Cross/Crossed/Uncross/Uncrossed procs at the AM level now make the AM crossing over the thing send a signal with the thing it crossed over as an argument. 10 days to figure out that simple fucking solution that has been staring me in the face this entire time. No need to do this check on Moved() anymore.